app/src/main/java/com/example/iphonesearchapi/adapter/ItunesAdapter.kt

```diff
@@ -23,8 +23,8 @@ class ItunesAdapter(var itunesList: List<Result>) :
         val value = itunesList[position]
         holder.textView.text = value.trackName
         holder.textView.setOnClickListener{view->
-            Navigation.findNavController(view).navigate(
-                R.id.action_searchFragment2_to_detailFragment
+            view.findNavController().navigate(
+                SearchFragmentDirections.actionSearchFragment2ToDetailFragment(value.trackId,value.trackName)
```

Comment:
findNavController() can return null if it doesn't find one. Maybe we can use the null safe operator here.
```suggestion
            view.findNavController()?.navigate(
                SearchFragmentDirections.actionSearchFragment2ToDetailFragment(value.trackId,value.trackName)
```